### ISSUE TITLE
More sinks to consider

### ISSUE THREAD

**Issue by christiankrug** [SAP/project-foxhound#144]
Here are a few sinks that seem to have gone unnoticed by Foxhound so far:

1. `src` seems to be a sink but `srcdoc` does not.
```
<html>
<body>
<iframe id="myframe"></iframe>
</body>
<script>
document.querySelector("#myframe").srcdoc = document.location.hash;
</script>
</html>
```
You can trigger this with the location.hash `#<img src='x' onerror='alert(1)'/>`.

2. `textContent`, a property that is usually safe, is not safe in certain scenarios, like when applying it to script elements.
```
<html>
<body>
<div id="test"></div>
</body>
<script>
const element = document.createElement('script');
element.textContent = document.location.hash.substring(1);
document.body.appendChild(element);
</script>
</html>
```
You can trigger this with the location.hash `#alert(1)`.

3. `createContextualFragment` is not defined as a sink. Consider this:
```
<html>
<body>
<script>
const myFragment = document.createRange().createContextualFragment(document.location.hash);
document.body.appendChild(myFragment);
</script>
</body>
</html>
```
You can trigger this with the location.hash `#<script>alert(1)</script>`.

4. Form actions:
```
<html>
<body>
  <form id="form" action="submit.php"></form>
</body>
<script>
document.querySelector("#form").action = document.location.hash.substring(1);
document.querySelector("#form").submit();
</script>
</html>
```
You can trigger this with the location.hash `#javascript:alert(1)`.

**Comment by leeN** [comment]
Thanks for the report!

The missing ones indeed seem like an oversight on our end, there are too many sinks in JS/HTML heh.

I'll have a look whether I find the time this week to add these sinks or is this already on your todo list @tmbrbr?

**Comment by tmbrbr** [comment]
@leeN please feel free to add the missing sinks!

They are on my To Do List, but quite far down at the moment! 😉

**Comment by tmbrbr** [comment]
Implemented in #149
